app/js/lib/foxbox/services/base.js

```diff
@@ -112,34 +114,72 @@ export default class BaseService extends EventDispatcher {
   /**
    * Setups value watcher for the getter matching specified selector.
    *
-   * @param {Object} selector Selector to match getter which value we would like
-   * to watch.
+   * @param {string} alias Watcher alias to match getter which value we would
+   * like to watch.
    * @param {function} handler Function to be called once getter value changes.
    */
-  watch(selector, handler) {
+  watch(alias, handler) {
```

Comment:
I like the watcher API and I like the we explicitly have to declare them.
As you say, let's experiment with them and see how it goes.